src/Agent.Worker/TestResults/NunitResultReader.cs

```diff
@@ -267,10 +243,94 @@ private List<TestCaseResultData> FindTestCaseNodes(XmlNode startNode, string hos
             return results;
         }
 
-        public bool AddResultsFileToRunLevelAttachments
+        public string InnerTestSuiteNodeName { get; set; }
+        public string TestCaseNodeName { get; set; }
+        public string RootNodeName { get; set; }
+
+    }
+
+
+    public interface INUnitResultsXmlReader
+    {
+        TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments);
+    }
+
+    internal class NUnit2ResultsXmlReader: NUnitResultsXmlReader, INUnitResultsXmlReader
+    {
+        public new TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments)
+        {
+            this.InnerTestSuiteNodeName = "results/test-suite";
+            this.TestCaseNodeName = "results/test-case";
+            this.RootNodeName = "test-results";
+
+            return base.GetTestRunData(filePath, doc, testResultsNode, runContext, addResultsAsAttachments);
+        }
+    }
+
+    internal class NUnit3ResultsXmlReader: NUnitResultsXmlReader, INUnitResultsXmlReader
+    {
+        public new TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments)
         {
-            get;
-            set;
+            this.InnerTestSuiteNodeName = "./test-suite";
+            this.TestCaseNodeName = "./test-case";
+            this.RootNodeName = "test-run";
+
+            return base.GetTestRunData(filePath, doc, testResultsNode, runContext, addResultsAsAttachments);
```

Comment:
Nunit 2.6 and before, the output file (results.xml) did not have 'start-time' attribute for each test suite. So we blindly added all the durations. But Nunit 3.0 has the 'start-time' attribute for each test suite. We need to have different implementation of GetTestRunData for Nunit 3.0 to handle results for parallel runs.